[KennaV2] Enhancement 2 New Commands (#32032)
demisto/content: Packs/Kenna/Integrations/KennaV2/README.md

Use the Kenna v2 integration to search and update vulnerabilities, schedule a run connector, and manage tags and attributes.
This integration was integrated and tested with version 1.0 of Kenna.

Some changes have been made that might affect your existing content. 

> If you are upgrading from a previous version of this integration, see [Breaking Changes](#breaking-changes-from-the-previous-version-of-this-integration-kenna-v2).

Reviewer: I don't see a breaking changes topic in this file.

> Some changes have been made that might affect your existing content. 
> If you are upgrading from a previous version of this integration, see [Breaking Changes](#breaking-changes-from-the-previous-version-of-this-integration-kenna-v2).

Reviewer 2: ```suggestion
```

| Kenna.Vulnerabilities.Score | Number | The vulnerability score. | 
| Kenna.Vulnerabilities.ServiceTicket.DueDate | Date | The service ticket due date. | 
| Kenna.Vulnerabilities.ServiceTicket.ExternalIdentifier | String | The service ticket external identifier. | 

> | Kenna.Vulnerabilities.ServiceTicket.Status | String | The segit rvice ticket status. | 

Reviewer: ```suggestion
| Kenna.Vulnerabilities.ServiceTicket.Status | String | The service ticket status. | 
```

#### Command example

```!kenna-search-fixes limit=2 id=dummy to_context=True```

#### Context Example

```json
{
    "Kenna": {
        "Fixes": {
            "Assets": [
                {
                    "DisplayLocator": "0.0.0",
                    "ID": 0,
                    "Locator": "0.0.0",
                    "PrimaryLocator": "ip_address"
                }
            ],
            "Category": null,
            "CveID": [
                "CVE-2015-0000"
            ],
            "ID": 0,
            "LastUpdatedAt": "2019-10-24T19:13:29.000Z",
            "MaxScore": 100,
            "Title": "CVE-2015-0000",
            "VulnerabilityCount": 1
        }
    }
}
```

#### Human Readable Output

>CVE-2015-0000

>#### ID: 0

>1 vulnerabilities affected

>#### Diagnosis:

>   Related CVE IDs:   CVE-2015-0000  
>&nbsp;

### kenna-update-asset

***
Updates the attributes of a single asset.

#### Base Command

`kenna-update-asset`

#### Input

| **Argument Name** | **Description** | **Required** |
| --- | --- | --- |
| id | The ID of the asset to update. | Required | 
| notes | Notes about the asset. | Required | 

> | inactive | Whether to inactivate the asset. Possible values are: true, false. | Optional | 

Reviewer: ```suggestion
| inactive | Whether to deactivate the asset. Possible values are: true, false. | Optional | 
```

| Kenna.ConnectorsList.Running | Boolean | The running connector. | 
| Kenna.ConnectorsList.Host | String | The connector host. | 

#### Command example

```!kenna-get-connectors```

#### Context Example

```json
{
    "Kenna": {
        "ConnectorsList": [
            {
                "Host": null,
                "ID": 0,
                "Name": "XML",
                "Running": false
            },
            {
                "Host": null,
                "ID": 1,
                "Name": "Generic",
                "Running": false
            }
        ]
    }
}
```

#### Human Readable Output

>### Kenna Connectors

>|Host|ID|Name|Running|
>|---|---|---|---|
>|  | 0 | XML | false |
>|  | 1 | Generic | false |

### kenna-search-assets

***
Searches for assets.

#### Base Command

`kenna-search-assets`

#### Input

| **Argument Name** | **Description** | **Required** |
| --- | --- | --- |
| id | The asset ID to search for. | Optional | 
| hostname | The hostname of the asset to search for. | Optional | 
| min-score | The minimum vulnerability score for which to return vulnerabilities. | Optional | 
| tags | The tags by which to search. | Optional | 

> | limit | The maximum number of vulnerabilities to return.Default is 500. | Optional | 

Reviewer: ```suggestion
| limit | The maximum number of vulnerabilities to return. Default is 500. | Optional | 
```

| Kenna.Assets.Owner | String | The asset owner. | 
| Kenna.Assets.Priority | Number | The asset priority. | 
| Kenna.Assets.Notes | String | Notes of current asset. | 

> | Kenna.Assets.OperatingSystem | String | Operating system of asset. | 

Reviewer 2: ```suggestion
| Kenna.Assets.OperatingSystem | String | Operating system of the asset.| 
```

> | Kenna.Assets.OperatingSystem | String | Operating system of asset. | 

Reviewer 2: ```suggestion
| Kenna.Assets.OperatingSystem | String | Operating system of the asset. | 
```

>|AssetsWithTagsReset|AutoclosedScannerVulnerabilities|AutoclosedVulnerabilities|ClosedScannerVulnerabilities|ClosedVulnerabilities|CreatedScannerVulnerabilities|EndTime|FailedPayload|ID|ProcessedAssets|ProcessedPayload|ProcessedScannerVulnerabilities|ReopenedScannerVulnerabilities|ReopenedVulnerabilities|StartTime|Success|TotalPayload|UpdatedScannerVulnerabilities|
>|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|
>| 0 | 0 | 0 | 0 | 0 | 10 | 2019-10-24T19:13:36.000Z | 0 | 111111 | 10 | 10 | 10 | 0 | 0 | 2019-10-24T19:02:02.000Z | true | 10 | 0 |
### kenna-inactivate-asset

***
Inactivates a single asset.

#### Base Command

`kenna-inactivate-asset`

#### Input

| **Argument Name** | **Description** | **Required** |
| --- | --- | --- |
| asset_id | The ID of the asset to inactivate. | Required | 
| notes | Notes about the asset. | Required | 

#### Context Output

There is no context output for this command.
### kenna-search-assets-by-external-id

***

> Search assets by external id.

Reviewer: ```suggestion
Search assets by external ID.
```

#### Base Command

`kenna-search-assets-by-external-id`

#### Input

| **Argument Name** | **Description** | **Required** |
| --- | --- | --- |

> | external_id | The external_id of the asset. | Required | 
> | to_context | Posibility of get data in context. Possible values are: true, false. Default is false. | Optional | 
> | limit | The maximum number of assets to return. The default value is 500. Default is 500. | Optional | 

Reviewer: ```suggestion
| external_id | The external ID of the asset. | Required | 
| to_context | Whether to put data in context. Possible values are: true, false. Default is false. | Optional | 
| limit | The maximum number of assets to return. Default is 500. | Optional | 
```

#### Context Output

| **Path** | **Type** | **Description** |
| --- | --- | --- |
| Kenna.Assets.ID | Number | The asset ID. | 
| Kenna.Assets.Notes | String | Notes of current asset. | 

> | Kenna.Assets.VulnerabilitiesCount | unknown | Count of Vulnerabilities of current asset. | 

Reviewer: ```suggestion
| Kenna.Assets.VulnerabilitiesCount | unknown | Count of vulnerabilities of current asset. | 
```

| Kenna.Assets.Hostname | unknown | Hostname of current asset. | 
| Kenna.Assets.Score | unknown | Score of current asset. | 

> | Kenna.Assets.IpAddress | unknown | Ip of current asset. | 

Reviewer: ```suggestion
| Kenna.Assets.IpAddress | unknown | IP of current asset. | 
```
